api/views.py

```diff
@@ -0,0 +1,185 @@
+import os
+import shutil
+import tarfile
+import traceback
+
+from django.views.decorators.http import require_http_methods
+from django.http.multipartparser import MultiPartParser
+from django.http import (
+    JsonResponse,
+    HttpResponseBadRequest,
+)
+from django.views.decorators.csrf import csrf_exempt
+
+from .utils import *
+
+
+@csrf_exempt
+@require_http_methods(['POST', 'PUT', 'DELETE'])
+def course_handler(request):
+    """
+    Upload/Update/Delete a course
+    """
+    # Verify the jwt token
+    try:
+        auth = jwt_auth(request)
+    except:
+        return HttpResponseBadRequest("JWT authentication failed!")
+
+    # Check the payload
+    if ('sub' not in auth) or (not auth['sub'].strip()):
+        return HttpResponseBadRequest("Invalid payload")
+
+    # The absolute path of the course in mooc-grader
+    course_path = settings.COURSES_PATH
+    course_directory = os.path.join(course_path, auth['sub'].strip())
+
+    message = None  # The message in successful response
+
+    if request.method == 'POST' or request.method == 'PUT':
+        # POST: upload a new course  PUT: upload a new course / update an existing course
+
+        if request.method == 'POST':
+            # The course already exists
+            if os.path.exists(course_directory):
+                return JsonResponse({'error': "The course has been uploaded"}, status=400)
+
+        # parse data
```

Comment:
This parsing logic is only required for `PUT`. Django handles this for `POST` and the data should be available in the `request.POST` and `request.FILES`

Reply:
I think for this api the `POST` should be enough. Use it for new and existing course.